odoo/addons/base/models/ir_ui_menu.py

```diff
@@ -314,3 +315,44 @@ def _get_menuitems_xmlids(self):
             menu.res_id: menu.complete_name
             for menu in menuitems
         }
+
+    @api.model
+    @tools.ormcache()
+    def _get_model_to_menu_root_info(self):
+        """ Get intermediate cached data useful to determine the best root menu for a given model.
+        The mapping is built from the models associated with ir.actions.act_window linked to menu items and propagated
+        to the root menus.
+        :return: nested dicts that maps res_model as str -> menu_id as int -> (menu_root_id, order_idx) as (int, int)
+          order_idx is a heuristic number that can be used to choose the best candidate (less is preferred)
+        """
+        context = {'ir.ui.menu.full_list': True}
+        menus = self.with_context(context).search([]).sudo()
+
+        # Build a nested dict that maps menu action model to menu root id while keeping the menu.id linked to the action
+        model_to_menu_id_to_menu_root_info = defaultdict(dict)
+        for menu in menus:
+            action = menu.action
+            if action and action.type == 'ir.actions.act_window':
+                order_idx = 0 if len(re.findall(MODEL_TO_MENU_ACTION_VIEW_MODE_PREFERRED, action.view_mode)) > 0 else 5
+                # number of : in context =~ number of variable in the context (dictionary)
+                order_idx += action.context.count(':') if action.context else 0
+                # number of , /3 in domain =~ number of condition
+                order_idx += action.domain.count(',')/3 if action.domain else 0
```

Comment:
Shouldn't you do a integer division ?
```suggestion
                order_idx += action.domain.count(',') // 3 if action.domain else 0
```

Reply:
Yes, it may be a little bit more accurate (it is any way an heuristic).
I will change as (action.domain.count(',')+1)//3. The +1 is needed as [('x','=','y')] will gives 0 otherwise.